File: 0125-valid-palindrome/0125-valid-palindrome.cs
public class Solution {
    public bool IsPalindrome(string s) {
        StringBuilder temp = new StringBuilder();
        for(int i = 0 ; i < s.Length; i++){
            char ch = char.ToLower(s[i]);
            if(ch >='a' && ch <='z')temp.Append(ch);
            if(ch >='0' && ch <='9')temp.Append(ch);
        }
        int  k = 0 , j = temp.Length-1;
        while(k<=j){
            if(temp [k] != temp[j]){
                return false;
            }
            k++;
            j--;
        }
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int i = 0 , j = s.size()-1;
        
        while(i<=j){
             s[i] = tolower(s[i]);
             s[j] = tolower(s[j]);
            if(s[i] >= 'a'  && s[i] <='z' && s[j] >='a' && s[j]<='z'){
                if(s[i] != s[j])return false;
                i++,j--;
                continue;
            }
            if(s[i] >= '0'  && s[i] <='9' && s[j] >='0' && s[j]<='9'){
                if(s[i] != s[j])return false;
                i++,j--;
                continue;
            }if(s[i] >= '0'  && s[i] <='9' && s[j] >= 'a' && s[j] <='z'){
                return false;
            }
            if(s[i] >= 'a'  && s[i] <='z' && s[j] >= '0' && s[j] <='9'){
                return false;
            }
            if(s[i] >= 'a'  && s[i] <='z'){
                j--;continue;
            }if(s[j] >= '0' && s[j] <='9'){
                i++;continue;
            }if(s[i] >= '0'  && s[i] <='9'){
                j--;continue;
            }if(s[j] >= 'a' && s[j] <='z'){
                i++;continue;
            }
            else 
            i++,j--;
        }
        return true;
    }
};